shell/common/rasterizer.cc

```diff
@@ -176,74 +189,71 @@ GrDirectContext* Rasterizer::GetGrContext() {
   return surface_ ? surface_->GetContext() : nullptr;
 }
 
-flutter::LayerTree* Rasterizer::GetLastLayerTree() {
-  return last_layer_tree_.get();
+flutter::LayerTree* Rasterizer::GetLastLayerTree(int64_t view_id) {
+  auto found = last_successful_tasks_.find(view_id);
+  if (found == last_successful_tasks_.end()) {
+    return nullptr;
+  }
+  return found->second.layer_tree.get();
 }
 
 void Rasterizer::DrawLastLayerTree(
     std::unique_ptr<FrameTimingsRecorder> frame_timings_recorder) {
-  if (!last_layer_tree_ || !surface_) {
+  if (last_successful_tasks_.empty() || !surface_) {
     return;
   }
-  RasterStatus raster_status = DrawToSurface(
-      *frame_timings_recorder, *last_layer_tree_, last_device_pixel_ratio_);
+  std::list<LayerTreeTask> tasks;
+  for (auto& [view_id, task] : last_successful_tasks_) {
+    tasks.push_back(std::move(task));
```

Comment:
Is moving correct here? `LayerTreeTask` does not have a move constructor, I would expect this to copy instead.

Reply:
Copying a unique pointer won't do what you want - moving should be right - but we should delete the copy/assign functionality of this struct.